### PR TITLE
Add definition of ISA circuits

### LINKED ISSUE #923: Emphasise ISA circuits more prominently in the transpile section
ISA circuits are circuits that match the target hardware definition. That is, it only consists of gates understood by the target hardware (basis gates), and any multi-qubit gates needed to obey any connectivity constraints (coupling map). The qiskit transpiler can convert abstract circuits to ISA circuits, and can also do more to optimise circuits beyond converting them to ISA format. circuits must be converted to ISA format before being submitted to runtime.

The transpile section docs should be updated to explain the facts above. I recommend including an intro paragraph in the introduction or one of the earlier pages defining ISA circuits, as well as shoutouts whenever we talk about the layout and translation stages (it could link back to that original definition paragraph). also open to other suggestions

### PR DESCRIPTION
**Pull request by kaelynj**
Closes #923 

Adds new subsection to introduction briefly defining ISA circuits and links to it from the Layout and Translation transpiler stage sections.

### CODE REVIEW

File: docs/transpile/index.mdx [hunk 1]
```diff
@@ -9,10 +9,15 @@ description: Introduction to transpiling quantum circuits in the Qiskit SDK.
...
 
 Find more information about the pass manager stages in the [Transpiler stages](transpiler-stages) topic.
 
+
+### Instruction Set Architecture
+In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA will consist only of gates which can be executed on the target hardware (i.e. the target's basis gates) and will only contain two-qubit gates according to the constraints specified by the connectivity of the hardware (i.e. the target's coupling map).  Circuits which follow the ISA of a particular backend is also a requirement when submitting jobs to one of IBM Quantum's backends through Qiskit IBM Runtime.
```
Comment (beckykd) [inline docs/transpile/index.mdx:18]:
```suggestion
In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA consist only of gates that can be run on the target hardware (the target's basis gates) and contain only two-qubit gates, according to the constraints specified by the connectivity of the hardware (the target's coupling map).  When using Qiskit Runtime to submit a job to an IBM Quantum system, the circuits must adhere to the system's ISA. 
```

File: docs/transpile/transpiler-stages.ipynb [hunk 2]
```diff
@@ -39,7 +39,7 @@
    "metadata": {},
    "source": [
     "## Layout stage\n",
-    "The next stage involves the layout or connectivity of the backend a circuit will be sent to.  In general, quantum circuits are abstract entities whose qubits are \"virtual\" or \"logical\" representations of actual qubits used in computations.  To execute a sequence of gates, a one-to-one mapping from the \"virtual\" qubits to the \"physical\" qubits in an actual quantum device is necesary.  This mapping is stored as a `Layout` object.\n",
+    "The next stage involves the layout or connectivity of the backend a circuit will be sent to.  In general, quantum circuits are abstract entities whose qubits are \"virtual\" or \"logical\" representations of actual qubits used in computations.  To execute a sequence of gates, a one-to-one mapping from the \"virtual\" qubits to the \"physical\" qubits in an actual quantum device is necesary.  This mapping is stored as a `Layout` object and is part of the constraints defined within a backend's [ISA](./#instruction-set-architecture).\n",
```
Comment (beckykd) [inline docs/transpile/transpiler-stages.ipynb:42]:
```suggestion
    "The next stage involves the layout or connectivity of the backend a circuit will be sent to.  In general, quantum circuits are abstract entities whose qubits are \"virtual\" or \"logical\" representations of actual qubits used in computations.  To execute a sequence of gates, a one-to-one mapping from the \"virtual\" qubits to the \"physical\" qubits in an actual quantum device is necesary.  This mapping is stored as a `Layout` object and is part of the constraints defined within a backend's [instruction set architecture (ISA)](./#instruction-set-architecture).\n",
```

File: docs/transpile/index.mdx [hunk 3]
```diff
@@ -9,10 +9,15 @@ description: Introduction to transpiling quantum circuits in the Qiskit SDK.
...
+A central component of the Qiskit SDK, the transpiler is designed for modularity and extensibility. Its central goal is to write new circuit transformations (known as transpiler **passes**), and combine them with other existing passes, greatly reducing the depth and complexity of quantum circuits.  Which passes are chained together and in which order has a major effect on the final outcome. This pipeline is determined by the [`PassManager`](/api/qiskit/qiskit.transpiler.PassManager) and [`StagedPassManager`](/api/qiskit/qiskit.transpiler.StagedPassManager) objects.  The `StagedPassManager` orchestrates the execution of one or more `PassManagers` and determines the order in which they are executed, while the `PassManager` object is merely a collection of one or more passes.  Think of the `StagedPassManager` as the conductor in an orchestra, the `PassManagers` as the different instrument sections, and the `Pass` objects as the individual musicians.  In this way, you can compose hardware-efficient quantum circuits that let you execute utility-scale work while keeping noise manageable.
```
Comment (abbycross) [inline docs/transpile/index.mdx:12]:
```suggestion
A central component of the Qiskit SDK, the transpiler is designed for modularity and extensibility. Its main goal is to write new circuit transformations (known as transpiler **passes**), and combine them with other existing passes, greatly reducing the depth and complexity of quantum circuits.  Which passes are chained together and in which order has a major effect on the final outcome. This pipeline is determined by the [`PassManager`](/api/qiskit/qiskit.transpiler.PassManager) and [`StagedPassManager`](/api/qiskit/qiskit.transpiler.StagedPassManager) objects.  The `StagedPassManager` orchestrates the execution of one or more `PassManagers` and determines the order in which they are executed, while the `PassManager` object is merely a collection of one or more passes.  Think of the `StagedPassManager` as the conductor in an orchestra, the `PassManagers` as the different instrument sections, and the `Pass` objects as the individual musicians.  In this way, you can compose hardware-efficient quantum circuits that let you execute utility-scale work while keeping noise manageable.
```

Comment (abbycross) [reply docs/transpile/index.mdx:12]: two "central"s in consecutive sentences :)

File: docs/transpile/index.mdx [hunk 4]
```diff
@@ -9,10 +9,15 @@ description: Introduction to transpiling quantum circuits in the Qiskit SDK.
...
 
 Find more information about the pass manager stages in the [Transpiler stages](transpiler-stages) topic.
 
+
+### Instruction Set Architecture
+In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA consist only of gates that can be run on the target hardware (the target's basis gates) and contain only two-qubit gates, according to the constraints specified by the connectivity of the hardware (the target's coupling map).  When using Qiskit Runtime to submit a job to an IBM Quantum backend, the circuits must adhere to the backend's ISA.
```
Comment (abbycross) [inline docs/transpile/index.mdx:18]:
```suggestion
### Instruction set architecture
In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA consist only of gates that can be run on the target hardware (the target's basis gates) and contain only two-qubit gates, according to the constraints specified by the connectivity of the hardware (the target's coupling map).  When using Qiskit Runtime to submit a job to an IBM Quantum&trade; backend, the circuits must adhere to the backend's ISA. 
```

Comment (abbycross) [reply docs/transpile/index.mdx:18]: sentence case and trademark

**Review by abbycross — APPROVED** [review]
👍

File: docs/transpile/index.mdx (hunk 4 above)
Comment (blakejohnson) [inline docs/transpile/index.mdx:18]: > consist only of gates that can be run on the target hardware

The definition here is a little narrow if read as `(instructions in circuit)` \subset `(basis gates)`. Can we add something that makes it clear that ISA circuits may include other instructions supported by the backend? e.g. `measure`, `reset`, control-flow operations (`if`, `else`, `for`, etc.)

Comment (blakejohnson) [reply docs/transpile/index.mdx:18]: > contain only two-qubit gates, according to

the grammatical construction here also reads as if the circuit can only consist of two-qubit gates.

Comment (kaelynj) [reply docs/transpile/index.mdx]: Just made a few edits based on these suggestions.  Let me know what you think.

**Review by javabster — APPROVED** [review]
LGTM 🚀

**Pull request merged**